### PR TITLE
Fixes inability to define the field_order in jobs.

### PR DESCRIPTION
**Pull request by smk4664**
### Fixes: #459

Fixes the field_order Meta option to jobs.

I can add tests for these, I noticed none of the tests for Jobs tested any of the optional attributes (Name, Description, Commit_Default).  We might want to make that a followup ticket...

### CODE REVIEW

**Comment by smk4664** [comment]
Per Django Docs, anything not provided by field_order would be appended to the bottom per the default order that Django would decide.

https://docs.djangoproject.com/en/3.2/ref/forms/api/#django.forms.Form.field_order

I will see what I can do about tests.

File: nautobot/docs/additional-features/jobs.md [hunk 1]
```diff
@@ -87,6 +87,10 @@ class MyJob(Job):
         commit_default = False
 ```
 
+#### `field_order`
+
+A list representing the order your form fields should appear.
```
Comment (glennmatthews) [inline nautobot/docs/additional-features/jobs.md:92]:
```suggestion
A list of strings (field names) representing the order your form fields should appear. If not defined, fields will appear in order of their definition in the code.
```

File: nautobot/docs/release-notes/version-1.0.md [hunk 2]
```diff
@@ -184,6 +184,7 @@ The "Related Devices" table has been removed from the detailed Device view.
 ### Changed
 
 - [#423](https://github.com/nautobot/nautobot/pull/423) - Clarified reference to `/config_contexts/` folder in Git user guide
+- [#467](https://github.com/nautobot/nautobot/pull/467) - Fixes issue with jobs not following field_order
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-1.0.md:187]: IMO this belongs either in `Added` (if we are regarding this as a feature) or `Fixed` (if it's a bug fix). My vote would be for the former but I'll accept either. :-)

I'd prefer to reserve `Changed` for genuine behavior changes.

File: nautobot/extras/tests/test_jobs.py [hunk 3]
```diff
@@ -54,3 +56,53 @@ def test_job_fail(self):
...
+<input class="form-control form-control" id="id_var23" name="var23" placeholder="None" required type="text">
+<br><span class="helptext">I want to be second</span></td></tr>
+<tr><th><label for="id__commit">Commit changes:</label></th><td>
+<input checked id="id__commit" name="_commit" placeholder="Commit changes" type="checkbox">
+<br><span class="helptext">Commit changes to the database (uncheck for a dry-run)</span></td></tr>""",
+            )
+
+    def test_no_field_order(self):
+        """
+        Job test with field order.
```
Comment (glennmatthews) [inline nautobot/extras/tests/test_jobs.py:87]:
```suggestion
        Job test without field_order.
```

File: nautobot/docs/release-notes/version-1.0.md (hunk 2 above)
Comment (jvanderaa) [reply nautobot/docs/release-notes/version-1.0.md:187]: Should be in Fixed IMO as there is a reference to it in the examples section of Jobs.

Comment (smk4664) [reply nautobot/docs/release-notes/version-1.0.md:187]: I didn't look long enough to find fixed. It should go in fixed, the jobs example uses field_order.

File: nautobot/docs/release-notes/version-1.0.md [hunk 4]
```diff
@@ -192,6 +192,7 @@ The "Related Devices" table has been removed from the detailed Device view.
 - [#428](https://github.com/nautobot/nautobot/issues/428) - Fixed GraphQL error when handling ASNs greater than 2147483647
 - [#430](https://github.com/nautobot/nautobot/pull/430) - Fixed missing `ContentType` foreign keys in GraphQL
 - [#461](https://github.com/nautobot/nautobot/issues/461) - Fixed NAUTOBOT_DB_TIMEOUT read as string in default config
+- [#467](https://github.com/nautobot/nautobot/pull/467) - Fixes issue with jobs not following field_order
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-1.0.md:195]:
```suggestion
- [#467](https://github.com/nautobot/nautobot/pull/467) - Fixes issue with job forms not respecting `field_order`
```

**Review by glennmatthews — APPROVED** [review]
Thank you!

**Review by jathanism — APPROVED** [review]
